### DISCUSSION TITLE
Marlin 2.x + ESP3D + BTT UPS + Power Lost Recovery

### DISCUSSION THREAD

**Discussion by ghost** [luc-github/ESP3D#896] [Questions]
Hi there,
I'm currently using Octoprint, and I'm aware this UPS power loss recovery method do not work.
So I was thinking if I drop Octoprint and use the ESP3D, will that way work?

**Comment by luc-github** [comment]
what do you mean ?

**Reply by ghost** [reply]
I mean will the G-code M413 - Power-loss Recovery work?

**Reply by ghost** [reply]
can I enable this in Marlin and would it work if I use ESP3D?

![image](https://user-images.githubusercontent.com/102892004/222172642-d79665d1-cbf8-4599-a2c0-e20aeea0a41b.png)

**Reply by luc-github** [reply]
I am not sure to understand
You mention Marlin feature, this feature does not affect ESP3D but Marlin only

**Reply by ghost** [reply]
Let me ask this way,
When we upload a .gcode file, Octoprint stores it in the RPI and sends each G-code sequentialy to the printer firmware.
This way, if a power failure occures, Octoprint cannot resume a failed print upon power recovery as it does not know what was actualy printing when at the power outage.

Does ESP3D like octoprint, stores the .gcode file and sends each command to the printer?

**Reply by luc-github** [reply]
no , currently ESP3D does not stream gcode, on some configuration ESP3D upload file on Printer SD 
ESP3D is a front end

**Reply by ghost** [reply]
Alright! that's what I needed to confirm.
Thanks a lot for your help!

**Reply by luc-github** [reply]
Streaming GCODE is a feature request, it is planned on ESP32 Dual core : https://github.com/luc-github/ESP3D/issues/243
But it is still not implemented, once done, for sure such user case should be considered
